scala-api/src/main/scala/com/cloudera/livy/scalaapi/ScalaJobHandle.scala

```diff
@@ -24,10 +24,30 @@ import scala.util.Try
 import com.cloudera.livy.JobHandle
 import com.cloudera.livy.JobHandle.{Listener, State}
 
+/**
+ * A handle to a submitted job. Allows for monitoring and controlling of the running remote job.
+ *
+ * Wrapper over the java JobHandle of livy
+ *
+ * @constructor Creates a ScalaJobHandle
+ * @param jobHandle the java JobHandle of livy
+ */
 class ScalaJobHandle[T] private[livy] (jobHandle: JobHandle[T]) extends Future[T] {
 
+  /**
+   * Return the current state of the job.
+   */
   def state: State = jobHandle.getState()
 
+  /** When the job is completed, either through an exception, or a value,
+   *  apply the provided function.
+   *
+   *  If the job has already been completed,
+   *  this will either be applied immediately or be scheduled asynchronously.
+   *
+   *  $multipleCallbacks
```

Comment:
Are these expanded when you actually generate the docs? The documentation says the macro needs to be defined in the same source file, which is not the case here.